eq-author/src/App/questionPage/Design/QuestionPageEditor/withHandleTransitionEnd.js

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+
+const withHandleTransitionEnd = WrappedComponent => {
```

Comment:
This feels more of a util than a HoC.

Reply:
Why a util rather than HOC? I guess it doesn't manage any state...?

Reply:
Yeah, it doesn't create any other components, add lifecycle methods etc and so it's not really react specific at the moment. Other than its given a ref.

Reply:
I view HoCs as awkward ways of sharing logic and so are only really justified when they are necessary. Nothing this is doing requires it to be a HoC. If it was doing something with context etc that required it then it would seem fine otherwise I would prefer a simpler system of sharing logic - a util file.

Additionally this component exposes some very generally named functions that are true for how they are used but do not explain what they do.

For example we don't expose `onButtonClick` from `withCreatePage` we expose `onCreatePage` as that is what it is doing and how it gets called is the components responsibility.

The fact it has this smell as well makes me feel that a HoC is not the right way of sharing this logic.

Reply:
As discussed we will move this to a `util`.